src/components/modules/paste.ts

```diff
@@ -753,11 +751,11 @@ export default class Paste extends Module {
         needToReplaceCurrentBlock = isCurrentBlockInitial && BlockManager.currentBlock.isEmpty;
       }
 
-      if (needToReplaceCurrentBlock) {
-        BlockManager.replace(tool, data, settings);
-      } else {
-        BlockManager.insert(tool, data, settings);
-      }
+      BlockManager.insert({
```

Comment:
insert and replace are different actions, why they are joined?

Reply:
In several cases when we want to insert block, we actually replace it.
For example, if you paste into empty paragraph it should be replaced. So instead of adding if statement you can now pass the `replace` parameter

```
if (shouldReplace) {
  BlockManager.replace(...);
} else {
  BlockManager.insert(...);
}

// vs

BlockManager.insert({..., replace: shouldReplace});
```